### ISSUE TITLE
Throw an error if tech isn't in options

### ISSUE THREAD

**Issue by mmcc** [videojs/video.js#1998]
If you try and unwittingly override all global options, you'll end up in a tech-less state and an unusable player (with none of the options you wanted). Right now this just silently fails, so we should consider throwing an error of some sort if there isn't a tech available in the options.

For example: 

```js
videojs.options = { width: 300, height: 150, controls: true, preload: auto };
videojs('sweet-vid'); // unusable player because all options have been overridden.
```

We should probably also actually document using global options correctly while we're at it.

**Comment by gkatsev** [comment]
I feel like `videojs.options` shouldn't just be an object available to everything. It should be changed to be a setter/getter like components are.

**Comment by forbesjo** [comment]
I'll look into this.

**Comment by forbesjo** [comment]
Freeing this ticket up if someone else want to take it.

**Comment by dmlap** [comment]
@gkatsev sounds like you're suggesting a more significant change. How about opening up a different issue to discuss?

**Comment by gkatsev** [comment]
We want a `setGlobalOptions` and `getGlobalOptions` for setting the global options rather than just setting it on `videojs.options`. We can do the check in that function.

**Comment by dmlap** [comment]
Got this one.

**Comment by heff** [comment]
Closed by #2097
